src/tensor_ops/reshape_to/mod.rs

```diff
@@ -73,7 +75,29 @@ pub trait ReshapeTo: HasErr + HasShape {
     ) -> Option<Result<Self::WithShape<Dst>, Self::Err>>;
 }
 
-impl<S: Shape, E: Dtype, D: ReshapeKernel<E>, T: Tape<E, D>> ReshapeTo for Tensor<S, E, D, T> {
+impl<S: Shape, E: Unit, D: ReshapeKernel<E>> ReshapeTo for Tensor<S, E, D> {
+    fn try_reshape_like<Dst: Shape>(
+        self,
+        dst: &Dst,
+    ) -> Option<Result<Self::WithShape<Dst>, Self::Err>> {
+        (self.shape().num_elements() == dst.shape().num_elements()).then(|| {
+            if self.shape.strides() == self.strides {
+                Ok(Tensor {
+                    id: self.id,
+                    data: self.data,
+                    shape: *dst,
+                    strides: dst.strides(),
+                    device: self.device,
+                    tape: self.tape,
+                })
+            } else {
+                self.device.forward(dst, &self)
+            }
+        })
+    }
+}
+
+impl<S: Shape, E: Dtype, D: ReshapeKernel<E>> ReshapeTo for Tensor<S, E, D, OwnedTape<E, D>> {
```

Comment:
Hmm apparently this doesn't work because anything using a generic tape `T` would need to ensure that `Tensor<..., T>: ReshapeTo`

Reply:
The only thing I can think of to fix this is using a macro to generate implementations for everything but bool, which has a backward implementation of `unimplemented!()` or something

Reply:
Going to revert this specific change - I addressed this in #635